liferay-tango/liferay-portal
# LPS-126044 Use CanonicalURL in MVC Commands, instead className/classPK

**Motivation**

Now all the Analytics Reports endpoints receive a className/classPK and with that we obtain the canonicalURL. But we can pass the canonicalURL as a parameter and save some time for calculations.

Also this way we can avoid calling the LayoutDisplayPageObjectProvider in every MVCCommand

**Proposed Solution**

1. Add a new method getCanonicalURL to the AnalyticsReportsInfoItem api.
2. Implement this new method for Journal and Layout.
3. Use the new method getCanonicalURL in the MVC total_views, total_reads, views and reads MVC commands, instead className and classPK

**How to test it**

Integration has been provided with this pull. As no UX changes has been made in this PR, there are no manual test that could be executed in order to test the changes (but all the manual test we execute for the Content Performance Panel should work as they did before)
